DeviceTests/DeviceTests.Shared/Share_Tests.cs

```diff
@@ -0,0 +1,24 @@
+﻿using System;
+using System.Threading.Tasks;
+using Xamarin.Essentials;
+using Xunit;
+
+namespace DeviceTests
+{
+    public class Share_Tests
+    {
+        [Fact]
+        public async Task Share_NullText()
+        {
+            string text = null;
+            await Assert.ThrowsAsync<ArgumentNullException>(() => Share.RequestAsync(text));
+        }
+
+        [Fact]
+        public async Task Share_NullRequest()
```

Comment:
Consider adding tests  which actually validate the behaviour noticed in the ticket, i.e. share requests which have nested properties that are null.